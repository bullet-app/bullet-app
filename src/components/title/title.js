import React from 'react';

import PropTypes from 'prop-types';

import { Main, Text } from './title.styled';

const Title = ({ secondary, children }) => (
  <Main>
    <Text secondary={secondary}>{children}</Text>
  </Main>
);

Title.propTypes = { secondary: PropTypes.bool };

export default Title;
