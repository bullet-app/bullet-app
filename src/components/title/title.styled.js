import styled from 'styled-components/native';

import { colors, font } from '../../utils';

export const Main = styled.View``;

export const Text = styled.Text`
  color: ${colors.text};
  font-size: ${props =>
    props.secondary ? font.title.base : font.title.base * 2}px;
`;
