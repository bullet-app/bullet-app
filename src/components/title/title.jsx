import React from 'react';

import PropTypes from 'prop-types';

import { Main, Text } from './title.styled';

const Title = ({ secondary, children }) => (
  <Main secondary={secondary}>
    <Text>{children}</Text>
  </Main>
);

Title.propTypes = { secondary: PropTypes.bool };

export default Title;
