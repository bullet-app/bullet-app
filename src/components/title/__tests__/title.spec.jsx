import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Title from '../title';

import { Main, Text } from '../title.styled';

describe('Title', () => {
  let wrapper;

  afterEach(() => wrapper && wrapper.unmount());

  describe('rendering', () => {
    it('should not render null', () => {
      // Assert
      expect((wrapper = shallow(<Title>Hello</Title>)).exists()).toBeTruthy();
    });

    it('should render Main', () => {
      // Assert
      expect(
        (wrapper = shallow(<Title>Hello</Title>)).find(Main).exists()
      ).toBeTruthy();
    });

    it('should render Text', () => {
      // Assert
      expect(
        (wrapper = shallow(<Title>Hello</Title>)).find(Text).exists()
      ).toBeTruthy();
    });

    it('should add the prop secondary to Main', () => {
      // Assert
      expect(
        (wrapper = shallow(<Title secondary>Hello</Title>))
          .find(Main)
          .prop('secondary')
      ).toBeTruthy();
    });
  });

  describe('snapshots', () => {
    it('should match the title snapshot', () => {
      // Assert
      expect(renderer.create(<Title>Hello</Title>).toJSON()).toMatchSnapshot();
    });

    it('should match the sub title snapshot', () => {
      // Assert
      expect(
        renderer.create(<Title secondary>Hello</Title>).toJSON()
      ).toMatchSnapshot();
    });
  });
});
