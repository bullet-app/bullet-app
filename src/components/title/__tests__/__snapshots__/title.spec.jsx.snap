// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title snapshots should match the sub title snapshot 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    secondary={true}
    style={
      Array [
        Object {
          "color": "#4c4c4c",
          "fontSize": 18,
        },
        undefined,
      ]
    }
  >
    Hello
  </Text>
</View>
`;

exports[`Title snapshots should match the title snapshot 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    secondary={undefined}
    style={
      Array [
        Object {
          "color": "#4c4c4c",
          "fontSize": 36,
        },
        undefined,
      ]
    }
  >
    Hello
  </Text>
</View>
`;
